### ISSUE TITLE
Unnecessary lambda linter misses multi-line cases

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1567]
Follow-up to #1541 to catch more cases

**Comment by MichaelChirico** [comment]
Adding new test cases here so that we can avoid `commented_code_linter()` in the sources. These all currently fail (0 lints instead of 1), bug is fixed by addressing these.

```
 test_that("cases with braces are caught", {
   linter <- unnecessary_lambda_linter()
   expect_lint(
     "lapply(x, function(xi) { print(xi) })",
     "xxx",
     linter
   )
   
   expect_lint(
     trim_some("
       lapply(x, function(xi) {
         print(xi)
       })
     "),
     "xxx",
     linter
   )
   
   expect_lint(
     "lapply(x, function(xi) { return(print(xi)) })",
     "xxx",
     linter
   )
   
   expect_lint(
     trim_some("
       lapply(x, function(xi) {
         print(xi)
       })
     "),
     "xxx",
     linter
   )
   
   expect_lint(
     trim_some("
       lapply(x, function(xi) {
         return(print(xi))
       })
     "),
     "xxx",
     linter
   )
 })
```
